tasks/create_kvm.yml

```diff
@@ -2,32 +2,90 @@
 
 # Crear una KVM desde cero en el nodo Proxmox
 - name: Create a KVM from scratch in Proxmox node
-  proxmox_kvm:
-    node: "{{ node }}"
-    api_user: "{{ api_user }}"
-    api_host: "{{ api_host }}"
-    api_password: "{{ api_password }}"
-    description: "{{ description | default (omit) }}"
-    kvm: "{{ kvm | default (omit) }}"
-    ostype: "{{ ostype | default (omit) }}"
-    name: "{{ proxmox_hostname }}"
-    sockets: "{{ sockets | default (omit) }}"
-    cores: "{{ cores | default (omit) }}"
-    cpuunits: "{{ cpuunits | default (omit) }}"
-    cpulimit: "{{ cpulimit | default (omit) }}"
-    memory: "{{ memory | default (omit) }}"
-    timeout: "{{ pve_timeout }}"
-    balloon: "{{ balloon | default (omit) }}"
-    shares: "{{ shares | default (omit) }}"
-    vga: "{{ vga | default (omit) }}"
+  community.general.proxmox_kvm:
+    node: "{{ pve_node }}"
+    api_host: "{{ pve_api_host }}"
+    api_user: "{{ pve_api_user | default(omit) }}"
+    api_password: "{{ pve_api_password | default(omit) }}"
+    api_token_id: "{{ pve_api_token_id | default(omit) }}"
+    api_token_secret: "{{ pve_api_token_secret | default(omit) }}"
+    proxmox_default_behavior: "{{ pve_kvm_proxmox_default_behavior | default('compatibility') }}"
+    validate_certs: "{{ pve_kvm_validate_certs | default(omit) }}"
+    name: "{{ pve_hostname }}"
+    vmid: "{{ pve_kvm_vmid | default(omit) }}"
+    timeout: "{{ pve_kvm_timeout | default(omit) }}"
+    description: "{{ pve_kvm_description | default(omit) }}"
+    kvm: "{{ pve_kvm_hardware_virtualization | default(omit) }}"
+    ostype: "{{ pve_kvm_ostype | default(omit) }}"
+    sockets: "{{ pve_kvm_sockets | default(omit) }}"
+    cores: "{{ pve_kvm_cores | default(omit) }}"
+    cpu: "{{ pve_kvm_cpu | default(omit) }}"
+    cpuunits: "{{ pve_kvm_cpuunits | default(omit) }}"
+    cpulimit: "{{ pve_kvm_cpulimit | default(omit) }}"
```

Comment:
Yes, and the same is true for many of the variables defined in this section.

>  I think the default omission here is more coherent with the new behavior of the module than "forcing" default values in the defaults definition file.

It was one of the alternatives that I evaluated, although I ended up deciding to give a value (the same as with `proxmox_default_behavior: compatibility`) for these variables in `defaults/main.yml` to highlight their relevance, and precisely for that reason they are part of the section `Most important/used vars`.

In any case, to be coherent throughout the role, either we leave all variables undefined (commented in `defaults/main.yml`) or we leave this as it is now.

Either option is fine for me. What do you think?

Reply:
I would let all optional variables commented and documented, but with no default value defined by the role. 

The reason is that (although I didn't test yet - because it needs to change the code -, but I think, or I presume) "no definition for an optional variable", so "omitted parameter when calling the module", has a specific meaning: "let proxmox itslef use it's default values and behavior". And if we set default values, we have no way to omit such parameters when calling de module. 

Remember we already has an incident that was simple, but quite hard to identify and with serious consequences on host performance, because of some default values silently set to inadequate values by older versions of the module: although we set (old variable) `cores` to a value > 1, `cpulimit` was set to 1 by the module, which means such limit for the average use of CPU, and not the proxmox Web GUI default value of 0, which means no limit below the allocated cores.

If we set default values to these optional variables, I'm afraid we may run into the same kind of side effect these "hidden" values, no longer in the module, but now in the role.

Reply:
I agree to leave the variables commented as documentation but not to define a default value in the role. Made [here](https://github.com/UdelaRInterior/ansible-role-proxmox-create-kvm/commit/e04dda598ad97775dcfc59bc9b98a202e47666f8).